src/V3AstNodes.cpp

```diff
@@ -2095,7 +2108,8 @@ void AstClassOrPackageRef::dump(std::ostream& str) const {
 }
 AstNodeModule* AstClassOrPackageRef::classOrPackagep() const {
     AstNode* foundp = m_classOrPackageNodep;
-    while (auto* const anodep = VN_CAST(foundp, Typedef)) foundp = anodep->subDTypep();
+    if (auto* const anodep = VN_CAST(foundp, Typedef)) foundp = anodep->subDTypep();
+    if (auto* const anodep = VN_CAST(foundp, NodeDType)) foundp = anodep->skipRefp();
```

Comment:
I believe the while is needed, as there can be a typedef that references a typedef.   Also likewise a parameter type can point to a typedef, so you need a loop passing through either/both until resolved.

Reply:
I think the Typedef skipping is there because it's used in the early linking stage before we have all the Ref's in place.  But you might be right the skipRef is appropriate, it matters where this is used.  Probably the best thing to do is add a typedef pointing to typedef in your added test.  If it works, we know what's sufficient.